### PR TITLE
Support resource naming strategy on Content Library API

### PR DESCRIPTION
**Pull request by ericvmw**
Add resource naming strategy to Content Library resource APIs, and include source ID field to Content Library item resource API.

Testing Done:

make targets pass

### CODE REVIEW

**Review by dilyar85 — APPROVED** [review]
LGTM!

File: api/v1alpha1/contentlibrary_types.go [hunk 1]
```diff
@@ -44,6 +44,20 @@ type StorageBacking struct {
 	DatastoreID string `json:"datastoreID,omitempty"`
 }
 
+// ResourceNamingStrategy represents a naming strategy for item resources in a content library in vCenter.
+type ResourceNamingStrategy string
+
+const (
+	// ResourceNamingStrategyFromItemID indicates the naming strategy that generates the item resource name from the item
+	// identifier for items in a content library.
+	ResourceNamingStrategyFromItemID ResourceNamingStrategy = "FROM_ITEM_ID"
```
Comment (fstrudel) [inline api/v1alpha1/contentlibrary_types.go:53]: Maybe indicate this is the default?

**Comment by fstrudel** [comment]
@ericvmw there's no code to actually support the new naming strategy right? Only the definition of the API in this MR?

**Review by fstrudel — APPROVED** [review]

**Comment by ericvmw** [comment]
> @ericvmw there's no code to actually support the new naming strategy right? Only the definition of the API in this MR?

Yes, this change is only the API def, the code to support it will be in the VM image registry operator.

File: api/v1alpha1/contentlibrary_types.go (hunk 1 above)
Comment (ericvmw) [reply api/v1alpha1/contentlibrary_types.go:53]: It is commented in the ResourceNamingStrategy field on the library specs, can add a comment here too.

**Pull request merged**